[WIP] Basic linting on save

Krzysztof-Cieslak opened the pull request:
1. Extensions is activated on detecting Elm file.
2. I've added linting on file save using elm make (assuming elm is in PATH)
3. Marked as WIP as I hope to do some refactoring, bit it's working.

rojepp commented:
Cool! 

This doesn't work on OSX at the moment. We will probably have to locate the elm executable. I can do this later, since I'm using OSX.
It should also fail for files where the path has spaces?

Krzysztof-Cieslak commented:
OK, so, I've fixed paths with spaces, finished my refactoring ( added promise based exec process function which probably will be useful for also other features) and added flag which makes elm auto answer to all possible prompts (this can be discussed but elm make often suggest for example updating packages etc).

Not sure why it doesn't work on Mac, can't really test it. But maybe it was problem with elm process hanging on some questions not Mac problem. It should work if elm is in PATH

rojepp commented:
Thanks! I'll hopefully get it on the marketplace tomorrow.
